Suggested spelling correction seems off-base

In https://www.reddit.com/r/rust/comments/gpw2ra/how_is_the_rust_compiler_able_to_tell_the_visible/ I noticed this surprising spelling suggestion:

```rust
#![feature(non_ascii_idents)]

fn main() {
    let _ = 读文;
}
```

```console
error[E0425]: cannot find value `读文` in this scope
   --> src/main.rs:45:13
    |
45  |     let _ = 读文;
    |             ^^^^ help: a tuple variant with a similar name exists: `Ok`
```

To me `读文` and `Ok` don't seem like they would be similar enough to meet the threshold for showing such a suggestion. Can we calibrate this better for short idents?

For comparison, even `kO` doesn't assume you mean `Ok`.

```console
error[E0425]: cannot find value `kO` in this scope
  --> src/main.rs:45:13
   |
45 |     let _ = kO;
   |             ^^ not found in this scope
```

rustc 1.45.0-nightly (8970e8bcf 2020-05-23)

Mentioning @estebank who worked on suggestions most recently in #65421.

--- a/compiler/rustc_span/src/edit_distance.rs
+++ b/compiler/rustc_span/src/edit_distance.rs
@@ -188,7 +188,11 @@ fn find_best_match_for_name_impl(
         return Some(*c);
     }
 
-    let mut dist = dist.unwrap_or_else(|| cmp::max(lookup.len(), 3) / 3);
+    // `fn edit_distance()` use `chars()` to calculate edit distance, so we must
+    // also use `chars()` (and not `str::len()`) to calculate length here.
+    let lookup_len = lookup.chars().count();
+
+    let mut dist = dist.unwrap_or_else(|| cmp::max(lookup_len, 3) / 3);
     let mut best = None;
     // store the candidates with the same distance, only for `use_substring_score` current.
     let mut next_candidates = vec![];

--- a/tests/ui/suggestions/non_ascii_ident.rs
+++ b/tests/ui/suggestions/non_ascii_ident.rs
@@ -1,0 +1,4 @@
+fn main() {
+    // There shall be no suggestions here. In particular not `Ok`.
+    let _ = 读文; //~ ERROR cannot find value `读文` in this scope
+}

--- a/tests/ui/suggestions/non_ascii_ident.stderr
+++ b/tests/ui/suggestions/non_ascii_ident.stderr
@@ -1,0 +1,9 @@
+error[E0425]: cannot find value `读文` in this scope
+  --> $DIR/non_ascii_ident.rs:3:13
+   |
+LL |     let _ = 读文;
+   |             ^^^^ not found in this scope
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0425`.

